### ISSUE TITLE
Invocation as python module `python -m ruff` does not work properly

### ISSUE THREAD

**Issue by trottomv** [charliermarsh/ruff#2520]
When ruff is installed on system without virtualenv or in a different path than the default, the invocation as a python module does not work, and a sys exit occurs without errors printed on the console:

`$ pip install ruff`
or
`$ pip install --user --no-cache ruff`

```
$ ruff -V
ruff 0.0.238
```

```
$ python -m ruff check -v .
$ ruff check -v .
...
[2023-02-03][10:15:35][ruff::rules::isort::categorize][DEBUG] Categorized 'json' as StandardLibrary (KnownStandardL
[2023-02-03][10:15:35][ruff::rules::isort::categorize][DEBUG] Categorized 'django' as ThirdParty (KnownThirdParty)
[2023-02-03][10:15:35][ruff::rules::isort::categorize][DEBUG] Categorized 'django_drf_filepond' as ThirdParty (NoMa
[2023-02-03][10:15:35][ruff::rules::isort::categorize][DEBUG] Categorized 'django' as ThirdParty (KnownThirdParty)
[2023-02-03][10:15:35][ruff::rules::isort::categorize][DEBUG] Categorized 'django' as ThirdParty (KnownThirdParty)
...
```

```
$ find / -name "ruff"
/home/trotto/.local/lib/python3.11/site-packages/ruff
/home/trotto/.local/bin/ruff
```

**Comment by charliermarsh** [comment]
Thanks for filing :) I'll take a look today (hopefully).

**Comment by charliermarsh** [comment]
Although where possible I recommend `ruff` over `python -m ruff`.

**Comment by trottomv** [comment]
Thank to you @charliermarsh 

in fact it is absolutely not a blocking problem, but it seems internal to the `__main__.py`

`sysconfig.get_path("scripts")` returns the correct path in case of a virtualenv is active, but if `pip install` is run to a custom PATH at that point `python -m` would no longer work because the result of `sysconfig.get_path("scripts ")` is always `/usr/local/bin`

**Comment by charliermarsh** [comment]
I guess we could call `subprocess.check_call` and add the scripts directory to the front of the path?

**Comment by trottomv** [comment]
I consider quite important to avoid the sys exit without having an explicit error in console, it is not easy to predict the various edge cases and where the ruff binary can be installed, i would suggest something like this:

```
if __name__ == "__main__":                                                   
    ruff = Path(sysconfig.get_path("scripts")) / "ruff"                      
    if not ruff.is_file():                                                   
        ruff = Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff"   
        if not ruff.is_file():                                               
            raise FileNotFoundError(ruff)                                    
    sys.exit(os.spawnv(os.P_WAIT, ruff, [ruff, *sys.argv[1:]]))              
                                                                             
```

This should cover most cases, 
if you like it, I can gladly open a PR.

**Comment by charliermarsh** [comment]
Yeah that seems to work for me. Feel free to send a PR! Would be welcome!

**Comment by charliermarsh** [comment]
Closed by #2563.
